Python/pythonrun.c

```diff
@@ -1288,6 +1288,11 @@ run_eval_code_obj(PyThreadState *tstate, PyCodeObject *co, PyObject *globals, Py
         }
     }
 
+    if (globals == NULL) {
+        PyErr_SetString(PyExc_RuntimeError, "globals are NULL");
+        return NULL;
+    }
```

Comment:
Could you please move it up above L1277 and change L1278 to an `else` clause. Error handling should occur as early as possible.